pkg/build/controller/build/defaults/defaults.go

```diff
@@ -100,6 +100,12 @@ func (b BuildDefaults) applyPodDefaults(pod *v1.Pod) {
 			}
 		}
 	}
+
+	if pod.Spec.Tolerations == nil && len(b.config.Tolerations) != 0 {
+		for _, toleration := range b.config.Tolerations {
+			pod.Spec.Tolerations = append(pod.Spec.Tolerations, toleration)
```

Comment:
since we're calling this a defaulter, you should make sure the config.Toleration.Key does not exist in pod.Spec.Tolerations before appending it.  (there's no reason to expect it to today, but that could change someday).